particle-iot/particle-cli
# Automate Protected Devices to run `particle usb` commands

## Description

Support Protected Devices while running `particle usb xxx` commands
All of the `particle usb xxx` commands enable Protection on the device after the command is finished except these two:
- `particle usb reset`
- `particle usb dfu`

Also, added a fix for `particle usb cloud-status device-name` which was throwing an error due to an already existing bug when using `device-name`

## How to Test

Run `particle usb xxx` commands on Protected, Protected (Service Mode), and Open Devices.
Run them when the device is in normal mode and dfu mode. 
- Are you seeing the expected output?
- Are you seeing the device recover if it errors?
- Does it show helpful error messages?

## Related Issues / Discussions

https://github.com/particle-iot/particle-cli/pull/754
https://app.shortcut.com/particle/story/129067/add-generic-helpers-for-a-device-to-put-into-service-mode-and-back-to-perform-a-cli-operation



## Completeness
